appIndicator.js

```diff
@@ -648,4 +660,43 @@ class AppIndicators_IconActor extends St.Icon {
         this._updateIcon()
         this._updateOverlayIcon()
     }
+
+
+    _setOpacity() {
+        let opacityValue = this._settings.get_int('icon-opacity');
+
+        this.opacity = opacityValue;
+    }
+
+    _setSaturation() {
+        let desaturationValue = this._settings.get_double('icon-saturation');
+
+        let sat_effect = this.get_effect('desaturate');
+        if (!sat_effect) {
+            sat_effect = new Clutter.DesaturateEffect({});
+            this.add_effect_with_name('desaturate', sat_effect);
+        }
+        sat_effect.set_factor(desaturationValue);
+    }
+
+    _setBrightnessContrast() {
+        let brightnessValue = this._settings.get_double('icon-brightness');
+        let contrastValue = this._settings.get_double('icon-contrast');
+
+        let bright_effect = this.get_effect('brightness-contrast');
+        if (!bright_effect) {
+            bright_effect = new Clutter.BrightnessContrastEffect({});
+            this.add_effect_with_name('brightness-contrast', bright_effect);
+        }
+        bright_effect.set_brightness(brightnessValue);
+        bright_effect.set_contrast(contrastValue);
+    }
+
+    _setIconSize() {
+        let sizeValue = this._settings.get_int('icon-size');
+        if (sizeValue > 0)
+            this._iconSize = sizeValue;
+        else
+            this._iconSize = this._iconSize0;
+    }
```

Comment:
The code above is in `_setIconSize()`.

If it would call `_invalidateIcon()`, that calls `_updateIcon()` which in turn calls `_setIconSize()`, and we have an infinite loop ;)

All changes on the icons are triggered by a call to `_invalidateIcon()`, there is never a direct call to `_setIconSize()` (and the others), so there is no need to invalidate it afterwards. It is already invalidated if this function (or any `_set*()`) is called.